From c0880fbd8f99d914323cff63a51148fe86dbc22e Mon Sep 17 00:00:00 2001
From: Sergio Aguirre <saaguirre@ti.com>
Date: Mon, 26 Jan 2009 15:19:50 -0600
Subject: [PATCH 215/304] OMAP3ISP: Frontend: Fix output horizontal pixel count

The number of Horizontal pixels output is NPH + 1, So configure as
width - 1.

Signed-off-by: Vimarsh Zutshi <vimarsh.zutshi@nokia.com>
---
 drivers/media/video/isp/ispccdc.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/media/video/isp/ispccdc.c b/drivers/media/video/isp/ispccdc.c
index 598584e..6bcb688 100644
--- a/drivers/media/video/isp/ispccdc.c
+++ b/drivers/media/video/isp/ispccdc.c
@@ -1189,13 +1189,13 @@ int ispccdc_config_size(u32 input_w, u32 input_h, u32 output_w, u32 output_h)
 		isp_reg_writel(0, OMAP3_ISP_IOMEM_CCDC, ISPCCDC_VP_OUT);
 		if (ispccdc_obj.ccdc_inpfmt == CCDC_RAW) {
 			isp_reg_writel(0 << ISPCCDC_HORZ_INFO_SPH_SHIFT
-					| (ispccdc_obj.ccdcout_w
+					| ((ispccdc_obj.ccdcout_w - 1)
 					<< ISPCCDC_HORZ_INFO_NPH_SHIFT),
 					OMAP3_ISP_IOMEM_CCDC,
 					ISPCCDC_HORZ_INFO);
 		} else {
 			isp_reg_writel(0 << ISPCCDC_HORZ_INFO_SPH_SHIFT
-					| (ispccdc_obj.ccdcout_w
+					| ((ispccdc_obj.ccdcout_w - 1)
 					<< ISPCCDC_HORZ_INFO_NPH_SHIFT),
 					OMAP3_ISP_IOMEM_CCDC,
 					ISPCCDC_HORZ_INFO);
@@ -1234,7 +1234,7 @@ int ispccdc_config_size(u32 input_w, u32 input_h, u32 output_w, u32 output_h)
 					OMAP3_ISP_IOMEM_CCDC,
 					ISPCCDC_VP_OUT);
 		isp_reg_writel(0 << ISPCCDC_HORZ_INFO_SPH_SHIFT |
-					(ispccdc_obj.ccdcout_w <<
+					((ispccdc_obj.ccdcout_w - 1) <<
 					ISPCCDC_HORZ_INFO_NPH_SHIFT),
 					OMAP3_ISP_IOMEM_CCDC,
 					ISPCCDC_HORZ_INFO);
-- 
1.5.6

